atomic_reactor/build.py

```diff
@@ -54,37 +53,33 @@ def _ensure_not_built(self):
 
 
 class BuildResult(object):
-    def __init__(self, command_result, image_id=None):
-        """ when build fails, image_id is None """
-        self.command_result = command_result
+    def __init__(self, logs=None, fail_reason=None, image_id=None):
+        """
+        :param logs: iterable of log lines (without newlines)
+        :param fail_reason: str, description of failure or None if successful
+        :param image_id: str, ID of built container image
+        """
+        # must provide one, not both
+        assert bool(fail_reason) != bool(image_id), \
+            "Either fail_reason or image_id should be provided, not both"
+        self._logs = logs or []
+        self._fail_reason = fail_reason
         self._image_id = image_id
 
     @property
-    def image_id(self):
-        return self._image_id
-
-    def is_failed(self):
-        return self.command_result.is_failed()
-
-    @property
     def logs(self):
-        return self.command_result.logs
-
-
-class ExceptionBuildResult(object):
-    def __init__(self):
-        self._logs = traceback.format_exc()
+        return self._logs
 
     @property
-    def image_id(self):
-        return None
+    def fail_reason(self):
+        return self._fail_reason
 
     def is_failed(self):
-        return True
+        return self._fail_reason is not None
```

Comment:
well in docstring is specified, that both has to be str

anyway added another assert